src/core/server/elasticsearch/merge_config.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { merge } from '@kbn/std';
+import { Writable } from '@kbn/utility-types';
+import type { ElasticsearchClientConfig } from './client';
+
+type WritableConfig = Writable<ElasticsearchClientConfig>;
```

Comment:
`ElasticsearchClientConfig` / `ElasticsearchConfig` defines their properties as readonly, so I had to use a writable version here.